### PR TITLE
Fix: Remove dependence on HTML structure

### PR DESCRIPTION
**Pull request by kenny-wealth**
- Remove dependence on HTML structure
- Reset pre-loaded drop-downs on change

### CODE REVIEW

File: src/public-assets/js/dependent-dropdown.js [hunk 1]
```diff
@@ -15,15 +15,17 @@
...
      * Clears and reloads the child drop-down for a parent drop-down
      * @event change
      */
-    $(document).on('change', 'select[data-dependent-dropdown]', function () {
+    $(document).on('change', dropdown_selector, function () {
         var $element = $(this);
+
         // Remove a dependent drop-down child if it already exists
-        $element.next('.dependent-dropdown-child').remove();
+        $(dropdown_selector).slice ( $(dropdown_selector).index( $element ) + 1).remove();
```
Comment (olajideoye) [inline src/public-assets/js/dependent-dropdown.js:28]: @kenny-wealth Does this work for multiple levels of dependent dropdowns? It doesn't look that way to me at first glance

File: src/public-assets/js/dependent-dropdown.js [hunk 2]
```diff
@@ -15,15 +15,17 @@
...
 
         $.ajax({
```
Comment (olajideoye) [inline src/public-assets/js/dependent-dropdown.js:30]: @kenny-wealth the success function on for this request is still dependent on the HTML structure.

**Review by olajideoye — CHANGES_REQUESTED** [review]
@kenny-wealth I've gone through the changes. For the edit case, is it difficult to build the former HTML structure

File: src/public-assets/js/dependent-dropdown.js (hunk 1 above)
Comment (kenny-wealth) [reply src/public-assets/js/dependent-dropdown.js:28]: @olajideoye Yes, it strips off all other dropdowns after the current element

File: src/public-assets/js/dependent-dropdown.js (hunk 2 above)
Comment (kenny-wealth) [reply src/public-assets/js/dependent-dropdown.js:30]: @olajideoye Yes, it creates the HTML itself here `$element.after('<div class="dependent-dropdown-child"></div>')`

**Review by yemexx1 — APPROVED** [review]

File: src/public-assets/js/dependent-dropdown.js (hunk 1 above)
Comment (olajideoye) [reply src/public-assets/js/dependent-dropdown.js:28]: Okay. Great!

**Review by olajideoye — APPROVED** [review]

**Pull request merged**